File: apps/api/forms/Component.Form.Application/ComponentHandler/Default/DefaultHandler.cs
using Component.Form.Application.Helpers;
using Component.Form.Model;

namespace Component.Form.Application.ComponentHandler.Default;

public class DefaultHandler : IComponentHandler
{
    public string GetDataType()
    {
        return SafeJsonHelper.GetSafeType(typeof(string));
    }

    public bool IsFor(string type)
    {
        return String.IsNullOrEmpty(type) || 
            type.Equals("text") ||
            type.Equals("select") ||
            type.Equals("multilineText") ||
            type.Equals("radio") ||
            type.Equals("checkbox") ||
            type.Equals("yesno") ||
            type.Equals("email") ||
            type.Equals("phonenumber") ||
            type.Equals("fileupload");    
    }

    public async Task<List<string>> Validate(string name, dynamic data, List<ValidationRule> validationRules, bool repeating = false, string repeatKey = "", int repeatIndex = 0)
    {
        if (validationRules == null || validationRules.Count == 0)
        {
            return new List<string>();
        }
        
        return await ExpressionHelper.Validate(data, validationRules, repeatIndex);
    }
}

File: apps/web/runner/app/utils/componentHandler/defaultHandler.ts
import { ComponentHandler } from '@/app/utils/componentHandler/interfaces';

export class DefaultComponentHandler implements ComponentHandler {
    static IsFor(type: string): boolean {
        return type === 'text' ||
            type === 'select' ||
            type === 'multilineText' ||
            type === 'radio' ||
            type === 'checkbox' ||
            type === 'yesno' ||
            type === 'email' ||
            type === 'phonenumber' ||
            type === 'fileupload';
    }

    GetFromFormData(name: string, data: FormData): any {
        const formData = data.get(name);
        if (formData === undefined) {
            return '';
        }
        if (formData === null) {
            return '';
        }
        return formData;
    }

    GetFromSavedData(name: string, data: { [key: string]: any }): any {
        if (data[name] === undefined) {
            return '';
        }
        if (data[name] === null) {
            return '';
        }
        return data[name];
    }
}
